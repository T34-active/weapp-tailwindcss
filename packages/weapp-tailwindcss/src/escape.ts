export const weappTwIgnore = String.raw

export { replaceWxml as escape } from './wxml/shared'

export { isAllowedClassName } from '@weapp-core/escape'
