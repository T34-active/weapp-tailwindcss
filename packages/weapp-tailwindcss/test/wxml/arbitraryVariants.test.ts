import { templateHandler } from '#test/v2/wxml'
import { MappingChars2String } from '@weapp-core/escape'

function simpleHandler(str: string) {
  return templateHandler(str, {
    escapeMap: MappingChars2String,
  })
}

describe('arbitrary variants', () => {
  it('[&:nth-child(3)]:underline', () => {
    const res = simpleHandler('<li class="[&:nth-child(3)]:underline">{item}</li>')
    expect(res).toMatchSnapshot()
  })

  it('lg:[&:nth-child(3)]:hover:underline', () => {
    const res = simpleHandler('<li class="lg:[&:nth-child(3)]:hover:underline">{item}</li>')
    expect(res).toMatchSnapshot()
  })

  it('[&_p]:mt-4', () => {
    const res = simpleHandler('<div class="[&_p]:mt-4"></div>')
    expect(res).toMatchSnapshot()
  })

  it('[@supports(display:grid)]:grid', () => {
    const res = simpleHandler('<div class="flex [@supports(display:grid)]:grid"></div>')
    expect(res).toMatchSnapshot()
  })

  it('[@media(any-hover:hover){&:hover}]:opacity-100', () => {
    const res = simpleHandler('<button type="button" class="[@media(any-hover:hover){&:hover}]:opacity-100"></button>')
    expect(res).toMatchSnapshot()
  })

  it('[&_.u-count-down__text]:!text-red-400', () => {
    const res = simpleHandler('<view class="after:border-none after:content-[\'Hello_World\']">after:border-none</view>')
    expect(res).toMatchSnapshot()
  })
})

describe('simpleHandler', () => {
  it('[&:nth-child(3)]:underline', () => {
    const res = simpleHandler('<li class="[&:nth-child(3)]:underline">{item}</li>')
    expect(res).toMatchSnapshot()
  })

  it('lg:[&:nth-child(3)]:hover:underline', () => {
    const res = simpleHandler('<li class="lg:[&:nth-child(3)]:hover:underline">{item}</li>')
    expect(res).toMatchSnapshot()
  })

  it('[&_p]:mt-4', () => {
    const res = simpleHandler('<div class="[&_p]:mt-4"></div>')
    expect(res).toBe('<div class="_n_p_cmt-4"></div>')
  })

  it('[@supports(display:grid)]:grid', () => {
    const res = simpleHandler('<div class="flex [@supports(display:grid)]:grid"></div>')
    expect(res).toMatchSnapshot()
  })

  it('[@media(any-hover:hover){&:hover}]:opacity-100', () => {
    const res = simpleHandler('<button type="button" class="[@media(any-hover:hover){&:hover}]:opacity-100"></button>')
    expect(res).toMatchSnapshot()
  })

  it('[&_.u-count-down__text]:!text-red-400', () => {
    const res = simpleHandler('<view class="after:border-none after:content-[\'Hello_World\']">after:border-none</view>')
    expect(res).toBe('<view class="aftercborder-none afterccontent-_qHello_Worldq_">after:border-none</view>')
  })
})
