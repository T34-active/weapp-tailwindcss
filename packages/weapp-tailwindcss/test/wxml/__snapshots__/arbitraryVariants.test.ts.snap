// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`arbitrary variants > [&:nth-child(3)]:underline 1`] = `"<li class="_ncnth-childy3y_cunderline">{item}</li>"`;

exports[`arbitrary variants > [&_.u-count-down__text]:!text-red-400 1`] = `"<view class="aftercborder-none afterccontent-_qHello_Worldq_">after:border-none</view>"`;

exports[`arbitrary variants > [&_p]:mt-4 1`] = `"<div class="_n_p_cmt-4"></div>"`;

exports[`arbitrary variants > [@media(any-hover:hover){&:hover}]:opacity-100 1`] = `"<button type="button" class="_jmediayany-hoverchoveryznchoverz_copacity-100"></button>"`;

exports[`arbitrary variants > [@supports(display:grid)]:grid 1`] = `"<div class="flex _jsupportsydisplaycgridy_cgrid"></div>"`;

exports[`arbitrary variants > lg:[&:nth-child(3)]:hover:underline 1`] = `"<li class="lgc_ncnth-childy3y_chovercunderline">{item}</li>"`;

exports[`simpleHandler > [&:nth-child(3)]:underline 1`] = `"<li class="_ncnth-childy3y_cunderline">{item}</li>"`;

exports[`simpleHandler > [@media(any-hover:hover){&:hover}]:opacity-100 1`] = `"<button type="button" class="_jmediayany-hoverchoveryznchoverz_copacity-100"></button>"`;

exports[`simpleHandler > [@supports(display:grid)]:grid 1`] = `"<div class="flex _jsupportsydisplaycgridy_cgrid"></div>"`;

exports[`simpleHandler > lg:[&:nth-child(3)]:hover:underline 1`] = `"<li class="lgc_ncnth-childy3y_chovercunderline">{item}</li>"`;
